src/asmcup/genetics/Gene.java

```diff
@@ -10,16 +12,27 @@ public Gene(byte[] dna, int gen, float score) {
 		this.gen = gen;
 		this.score = score;
 	}
-	
+
+	@Override
 	public int compareTo(Gene other) {
 		float d = score - other.score;
 		
-		if (d == 0) {
+		if (Math.abs(d) < 0.00000001f) {

```

Comment:
Excuse me, but I still don't understand it.

You are relying on both floats to be equal.

Let's say you will increase score of Gene A by `0.7f` and score of Gene B by `0.37f` and then `0.33f`.
They will not compare to be equal (`0.37f + 0.33f != 0.7f`).

> For what we care about, they are only equal if they are literally the same object instance.
But you don't check for it. You check for score having the same binary representation.

I understand that you are creating an ordering. But it's arbitrary. 

After all I don't think it's worth fighting over this, as I don't seem to understand it anyway.
Additionally, I haven't looked into Genes and might be missing the crucial point.

I've already reverted this hours ago.